internal/fnruntime/runner.go

```diff
@@ -179,32 +179,35 @@ type FunctionRunner struct {
 
 func (fr *FunctionRunner) Filter(input []*yaml.RNode) (output []*yaml.RNode, err error) {
 	pr := printer.FromContextOrDie(fr.ctx)
+	var fnOutput strings.Builder
 	if !fr.disableCLIOutput {
-		pr.Printf("[RUNNING] %q", fr.name)
+		fnOutput.WriteString(fmt.Sprintf("[RUNNING] %q", fr.name))
 		if fr.displayResourceCount {
-			pr.Printf(" on %d resource(s)", len(input))
+			fnOutput.WriteString(fmt.Sprintf(" on %d resource(s)", len(input)))
 		}
-		pr.Printf("\n")
+		fnOutput.WriteString("\n")
 	}
 	t0 := time.Now()
 	output, err = fr.do(input)
 	if err != nil {
-		printOpt := printer.NewOpt()
-		pr.OptPrintf(printOpt, "[FAIL] %q in %v\n", fr.name, time.Since(t0).Truncate(time.Millisecond*100))
-		printFnResult(fr.ctx, fr.fnResult, printOpt)
+		fnOutput.WriteString(fmt.Sprintf("[FAIL] %q in %v\n", fr.name, time.Since(t0).Truncate(time.Millisecond*100)))
+		fnOutput.WriteString(printFnResult(fr.fnResult))
 		var fnErr *ExecError
 		if goerrors.As(err, &fnErr) {
-			printFnExecErr(fr.ctx, fnErr)
+			fnOutput.WriteString(printFnExecErr(fnErr))
+			pr.Printf(fnOutput.String())
```

Comment:
Why do we still need to print from here?

Reply:
This is a remnant from the previous implementation, which handled errors returned from the function differently from function execution errors. This PR removes the need to do a separate print, so I have updated the PR accordingly.